### PR TITLE
Add Akeyless Gateway Integration Tile

### PR DESCRIPTION
**Pull request by akljph**
### What does this PR do?

A brief description of the change being made with this pull request.

### Motivation

What inspired you to submit this pull request?

### Review checklist

- [ ] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [ ] Feature or bugfix has tests
- [ ] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

**Review by emarsha94 — CHANGES_REQUESTED** [review]

File: akeyless_gateway/manifest.json [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        "image_url": "images/akeyless_gw_dashboard.png",
+        "media_type": "image"
+      }
+    ],
+    "classifier_tags": [
+      "Category::Security",
+      "Category::Kubernetes",
+      "Supported OS::Linux",
+      "Supported OS::Windows",
+      "Supported OS::macOS"
```
[the remark this answers is not in the document]
Comment (akljph) [reply akeyless_gateway/manifest.json]: Done :)

File: akeyless_gateway/metadata.csv [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+akeyless.gw.system.cpu,gauge,,,,CPU utilization metrics,0,akeyless_gateway,,
+akeyless.gw.system.disk,gauge,,,,Disk I/O metrics,0,akeyless_gateway,,
+akeyless.gw.system.load,gauge,,,,CPU load metrics,0,akeyless_gateway,,
+akeyless.gw.system.memory,gauge,,,,Memory utilization metrics,0,akeyless_gateway,,
+akeyless.gw.system.network,gauge,,,,Network interface I/O metrics & TCP connection metrics,0,akeyless_gateway,,
+akeyless.gw.quota.current_transactions_number,gauge,,,,Current transaction number,0,akeyless_gateway,,
+akeyless.gw.quota.gw_admin_client_transactions,gauge,,,,Total transaction by an admin client,0,akeyless_gateway,,
+akeyless.gw.quota.total_transactions_limit,gauge,,,,Total transaction limit per hour,0,akeyless_gateway,,
+akeyless.gw.system.http_response_status_code,gauge,,,,Status of HTTP response,0,akeyless_gateway,,
+akeyless.gw.system.request_count,gauge,,,,Total number of requests,0,akeyless_gateway,,
```
Comment (alai97) [inline akeyless_gateway/metadata.csv:11]:
```suggestion
akeyless.gw.system.cpu,gauge,,,,CPU utilization metrics.,0,akeyless_gateway,,
akeyless.gw.system.disk,gauge,,,,Disk I/O metrics.,0,akeyless_gateway,,
akeyless.gw.system.load,gauge,,,,CPU load metrics.,0,akeyless_gateway,,
akeyless.gw.system.memory,gauge,,,,Memory utilization metrics.,0,akeyless_gateway,,
akeyless.gw.system.network,gauge,,,,Network interface I/O metrics and TCP connection metrics.,0,akeyless_gateway,,
akeyless.gw.quota.current_transactions_number,gauge,,,,Current transaction number.,0,akeyless_gateway,,
akeyless.gw.quota.gw_admin_client_transactions,gauge,,,,Total transaction by an admin client.,0,akeyless_gateway,,
akeyless.gw.quota.total_transactions_limit,gauge,,,,Total transaction limit per hour.,0,akeyless_gateway,,
akeyless.gw.system.http_response_status_code,gauge,,,,Status of the HTTP response.,0,akeyless_gateway,,
akeyless.gw.system.request_count,gauge,,,,Total number of requests.,0,akeyless_gateway,,
```

File: akeyless_gateway/README.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
+# Akeyless Gateway
+
+## Overview
+
+The Akeyless Platform is a unified secrets management system that enables you to store, protect, rotate, and dynamically create credentials, certificates, and encryption keys. Our platform supports several use cases, including managing static and dynamic credentials, certificate automation, encryption and digital signing, and zero-trust application access that secures remote access to your internal resources.
+
+This integration allows you to visualize and monitor performance of your [Akeyless Gateway](https://docs.akeyless.io/docs/api-gw). Telemetry metrics are sourced from the application and the runtime environment.
```
Comment (alai97) [inline akeyless_gateway/README.md:7]:
```suggestion
This integration allows you to visualize and monitor performance of your [Akeyless Gateway][2]. Telemetry metrics are sourced from the application and the runtime environment.
```

File: akeyless_gateway/README.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+## Setup
+
+Akeyless offers a unique Gateway, which adds an extra level of protection between your private network and the cloud. Acting as a SaaS extension of our core services, our stateless Gateway enables a transparent internal operation with a robust out-of-the-box mechanism to ensure service continuity and recovery without having to change any network infrastructure in order to work with your internal resources.
```
Comment (alai97) [inline akeyless_gateway/README.md:11]:
```suggestion
Akeyless offers a unique Gateway which adds an extra level of protection between your private network and the cloud. Acting as a SaaS extension of our core services, our stateless Gateway enables a transparent internal operation with a robust out-of-the-box mechanism to ensure service continuity and recovery without having to change any network infrastructure in order to work with your internal resources.
```

File: akeyless_gateway/README.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+### Prerequisites
+- An Akeyless Gateway either running or being deployed for the first time
+
+### Configure
+
+This integration works with one Gateway or multiple instances using the same API key. Metrics can be shown per `host` or `instance` in the **Akeyless GW** dashboard.
+
+#### For a Gateway running on Kubernetes
+
+To configure the Akeyless Gateway integration on a [Gateway running on K8s](https://docs.akeyless.io/docs/gateway-k8s):
```
Comment (alai97) [inline akeyless_gateway/README.md:24]:
```suggestion
To configure the Akeyless Gateway integration on a [Gateway running on K8s][3]:
```

File: akeyless_gateway/README.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+1. In your `values.yaml` file you use to deploy your Gateway on Kubernetes, under the `metrics` section, add the below configuration. Set the relevant API Key of your Datadog server, and the relevant [Datadog site][1] (i.e. `app.datadoghq.com`).
```
Comment (alai97) [inline akeyless_gateway/README.md:26]:
```suggestion
1. In your `values.yaml` file you use to deploy your Gateway on Kubernetes, under the `metrics` section, add the following configuration. Set the relevant API Key of your Datadog server, and the relevant [Datadog site][1] such as: `app.datadoghq.com`.
```

File: akeyless_gateway/README.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+        api:
+          key: "<Your Datadog API key>"
+          site: <Your Datadog server site>         
+    service:
+      pipelines:
+        metrics:
+          exporters: [datadog]
+```
+
+2. If you have not yet deployed the Gateway, continue with your installation as usual and run the following command when ready to deploy:
```
Comment (alai97) [inline akeyless_gateway/README.md:43]:
```suggestion
2. If you have not yet deployed the Gateway, continue with your installation as usual and run the following command when you are ready to deploy:
```

File: akeyless_gateway/README.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+
+3. If you are updating an existing Gateway on Kubernetes, run the following commands to update:
+
+```
+helm upgrade <your-gateway-name> akeyless/akeyless-api-gateway -f values.yaml
+```
+
+#### For a standalone Gateway running on Docker
+
+To configure the Akeyless Gateway integration on a [Standalone Gateway](https://docs.akeyless.io/docs/install-and-configure-the-gateway):
```
Comment (alai97) [inline akeyless_gateway/README.md:57]:
```suggestion
To configure the Akeyless Gateway integration on a [Standalone Gateway][4]:
```

File: akeyless_gateway/README.md [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+
+1. Create a local file called `otel-config.yaml` with the below configuration. Set the relevant API Key of your Datadog server, and the relevant [Datadog site][1] (i.e. `app.datadoghq.com`).
```
Comment (alai97) [inline akeyless_gateway/README.md:59]:
```suggestion
1. Create a local file called `otel-config.yaml` with the following configuration. Set the relevant API Key of your Datadog server, and the relevant [Datadog site][1] such as `app.datadoghq.com`.
```

File: akeyless_gateway/README.md [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+    api:
+      key: "<Your Datadog API key>"
+      site: <Your Datadog server site>
+service:
+  pipelines:
+    metrics:
+      exporters: [datadog]
+```
+
+2. If you have not yet deployed the Gateway, run the below command to spin up your Akeyless Gateway with the `ENABLE_METRICS=true` variable and mounting the `otel-config.yaml` file:
```
Comment (alai97) [inline akeyless_gateway/README.md:73]:
```suggestion
2. If you have not yet deployed the Gateway, run the following command to spin up your Akeyless Gateway with the `ENABLE_METRICS=true` variable and mount the `otel-config.yaml` file:
```

File: akeyless_gateway/README.md [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+```
+3. If you are updating an existing Gateway, use the same `Admin Access ID` and `Cluster Name` for the updated Gateway in order to retrieve the latest settings and data from the previously removed Docker instance:
+
+```
+docker run -d -p 8000:8000 -p 8200:8200 -p 18888:18888 -p 8080:8080 -p 8081:8081 -p 5696:5696 -e ADMIN_ACCESS_ID="p-xxxxxx" -e ADMIN_ACCESS_KEY="62Hu...xxx....qlg=" -e ENABLE_METRICS="true" -v $PWD/otel-config.yaml:/akeyless/otel-config.yaml --name <your-gateway-name> akeyless/base:latest-akeyless
+```
+
+### Validation
+
+Upon successful setup of the Gateway, go to the Metrics page on the Datadog platform, and filter the Akeyless metrics on the summary page.
```
Comment (alai97) [inline akeyless_gateway/README.md:86]: What link should we include here, perhaps a dashboard link?
```suggestion
Upon successful setup of the Gateway, go to the [**Metrics** page][5] on the Datadog site, and filter the Akeyless metrics on the summary page.
```

File: akeyless_gateway/README.md [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+
+## Data Collected
+
+### Metrics
+
+See [metadata.csv](/akeyless_gateway/metadata.csv) for a list of metrics provided by this integration.
```
Comment (alai97) [inline akeyless_gateway/README.md:92]:
```suggestion
See [metadata.csv][6] for a list of metrics provided by this integration.
```

File: akeyless_gateway/README.md [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+
+### Service Checks
+
+Akeyless Gateway does not include any service checks.
```
Comment (alai97) [inline akeyless_gateway/README.md:96]:
```suggestion
The Akeyless Gateway integration does not include any service checks.
```

File: akeyless_gateway/README.md [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+
+### Events
+
+Akeyless Gateway does not include any events.
```
Comment (alai97) [inline akeyless_gateway/README.md:100]:
```suggestion
The Akeyless Gateway integration does not include any events.
```

File: akeyless_gateway/README.md [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+
+## Support
+
+Need help? Contact [Akeyless Support](mailto:support@akeyless.io).
+
+
+[1]: https://docs.datadoghq.com/getting_started/site/
```
Comment (alai97) [inline akeyless_gateway/README.md:107]: Remember to replace [5] with a link to the "Metrics page on the Datadog Site that shows Akeyless metrics on the summary page".

```suggestion
[1]: https://docs.datadoghq.com/getting_started/site/
[2]: https://docs.akeyless.io/docs/api-gw
[3]: https://docs.akeyless.io/docs/gateway-k8s
[4]: https://docs.akeyless.io/docs/install-and-configure-the-gateway
[5]: TBD
[6]: https://github.com/DataDog/integrations-extras/blob/master/akeyless_gateway/metadata.csv
[7]: mailto:support@akeyless.io
```

File: akeyless_gateway/README.md [hunk 16]
```diff
@@ -0,0 +1,107 @@
...
+
+Akeyless Gateway does not include any service checks.
+
+### Events
+
+Akeyless Gateway does not include any events.
+
+## Support
+
+Need help? Contact [Akeyless Support](mailto:support@akeyless.io).
```
Comment (alai97) [inline akeyless_gateway/README.md:104]:
```suggestion
Need help? Contact [Akeyless Support][7].
```

**Comment by akljph** [comment]
@alai97 

> Some suggestions + reminder to replace a link to a "Metrics page in Datadog"!

Should be all done :)

**Review by emarsha94 — APPROVED** [review]

**Review by FlorentClarret — APPROVED** [review]

**Review by james-eichelbaum — APPROVED** [review]

**Pull request merged**